compiler/noirc_driver/src/contract.rs

```diff
@@ -41,6 +41,8 @@ pub struct CompiledContract {
 pub struct ContractFunction {
     pub name: String,
 
+    pub hash: u64,
+
```

Comment:
These shouldn't have hashes as we don't use them.

Reply:
In the next PR I got rid of the `Circuit` type in the `execute_cmd` of `noir-execute` in favour of mapping both programs and contract functions back to a `CompiledProgram` [here](https://github.com/noir-lang/noir/pull/7406/files#diff-efbce6b0629ea34dbdf888d453bc5c1371704981a7c2af869a56d39a329e56c8R80). 

I understand we don't use them at the moment in `nargo`, however the `ContractFunction` comes from a `CompiledProgram` and at that point it does have `hash`. I don't see why, if programs and functions come from the same source, we have to lose this information, when it can be handy if for nothing else than providing some extra metadata about a failed contract execution.

Reply:
I can imagine that a contract fails, we have a dump from Nico, and yes we can execute the artifact based on its path, but with the `hash` we can also compile the source from `nargo` and ascertain that we are dealing with the same code, before starting to modify it for debugging. 

I’m sure other hashes like the git hash of the contract repo can be added somewhere by the relevant teams to make this a non-issue, but also don’t see the downside of having it.

Reply:
I'd also argue that adding these fields, and the ability to deal with both kinds of artefacts in our CLIs with the minimum amount of special cases is a nice way to minimise the need to [do the same in Javascript](https://github.com/AztecProtocol/aztec-packages/blob/2e83891b1b1d080febe5f3eac5dfe51e5844cf06/noir-projects/noir-contracts/scripts/extractFunctionAsNoirArtifact.js#L27-L36). For example that script has to do what I did in this PR by setting `hash` to 0 and `names` to some non-empty array to be able to pass it to the profiler. If we do it ourselves, it's less likely that the profiler or the inspector will be surprised at the input.

Reply:
That's fair, there's no harm in keeping them.